### ISSUE TITLE
[EN][paladin] Spell] crusader strike/Divine Storm

### ISSUE THREAD

**Issue by Frainzil** [Atlantiss/BugTracker#2725]
http://www.wowhead.com/spell=35395/crusader-strike#changelog
http://www.wowhead.com/spell=53385/divine-storm#changelog
http://www.wowhead.com/spell=25956/sanctity-of-battle

As you know, with this talent the cooldown of crusader strike/Divine storm is affected by haste, so that the more haste you have, the lower the internal cd is of your crusader strike/Divine storm. I recently noticed that a lot of times during fights i click on my crusader strike whenever its ready, and i get the message 'Spell is not ready yet', even when i got my spell ready according to my spell bar/addons. I asked few of my guildy's to test the exact same thing, just cast crusader strike and see if you get the same problem, and they do .Now there is in my opinion only 1 posibility for that; I guess that the talent is just not working, though the cd of crusader strike is correct when you look at the spell information. So for example if you crusader strike has a 4.30sec cooldown, and you press it 0.05 sec later, it will tell you that your spell is not ready, all spells will ''reload'' and you lost a cast of 1 of your spells which is +- 0.500ms. If you stack this up over a whole fight it means you can lose up to 10% dmg on your crusader strike + 10% less holy power generated by using it.  It would be nice if this gets fixed soon cause its the main ability of a retri pala? :s

**Comment by Glare** [comment]
It's working, check with Bloodlust/Heroism. BUT it is somehow messed up because, as you said, it blocks our CS/DS very often. Maybe, just maybe, it's not working with haste we get from items? Because it seems to work good with BL/Hero and our Judgement(talented) steroid.

**Comment by Frainzil** [comment]
well its kinda hard to get items without 1 of them having any haste at all. So there will always be somewhat of a delay, even during tw/BL i notice sometime that there is the bug again, its true the global cd of CS/DS changes when using tw/BL but still it bugs from time to time cause of something which is blocking you from casting it.

**Comment by Kejmil** [comment]
well on 4.0.6 it was sanctify of battle which was causing this problems, now... no idea

**Comment by Glare** [comment]
Here are results of my quick testing.

1. With talent Sanctity of Battle and without haste from gear - problem still present.
2. Without talent Sanctity of Battle - problem still present.

So it looks like something is wrong with Crusader Strike and Divine Storm, not with talent itself.

**Comment by Frainzil** [comment]
Well i really have no idea what may possibly be wrong if its not the talent which is causing this bug. It is also not a problem to cast CS/DS at the same time as your melee attack ( and i keep getting the bug even if i'm not doing an auto attack at al lright before the cast so there is nothing really to 'block it').so it has to be something wrong with the cd of the spell itself, unafected by any talent or spell improving haste.

**Comment by Clarkiez96** [comment]
I noticed this during the Altairus Fight in the Vortex Pinnacle. If you have his red debuff the icon of your Crusader Strike appears to be ready but when it's clicked it comes up with 'Spell not ready yet', haven't really noticed anything outside of that though.

**Comment by Glare** [comment]
I don't know if it's Blizzlike or not but that's because haste debuffs affects(increase) your CS/DS cooldown.

**Comment by Frainzil** [comment]
Well after doing multiple fights, i noticed that it has practically nothing to do with melee swings blocking you from casting CS/DS nor that the talent is bugging the spell. So it means that just the scripts of the spell must be broken or sometihing else must be delaying the spellcast.

**Comment by Glare** [comment]
Indeed it has nothing to do with meele swings. You can test it on DS outside of combat.

**Comment by Clarkiez96** [comment]
Yes Glare it is blizzlike, the talent Sanctity of Battle makes haste reduce the cool down of CS and DS. I believe there is just a visual scripting problem, you can always fire off CS after its set cool down is over, but the Visual appears to not 'update' it's timer when our haste is increased/decreased

**Comment by Frainzil** [comment]
yea i also think its just something wrong with the spell cast and the execution of it. It isn't that the spell isn't ready, whenever you try to cast a different spell which isn't ready yet, the game tells you its not ready. BUT when you try this with cs whenever it is ready, you get the same note; your spell isn't ready yet, but instead of doing nothing, you still try to cast the spell. Something is just resetting the swing of your CS and it has nothing to do with any of your talents. Whenever you cast a spell, you have the standard waiting time before you can cast a new one and you see your spells ''reload'' so you can only cast 1 spell each ... time. Now when you do it with crusader strike, you can see the game starts that animation of reloading the spells, but after a really short amount of time, it resets. So you actually do cast the spell, something is just resetting the spell/blocking it from finishing the cast.

**Comment by Clarkiez96** [comment]
That's a pretty good observation, I just noticed that, hopefully that will help the staff fix the issue

**Comment by Frainzil** [comment]
Also i noticed something else, so whenever you are alone, and test your rotation on the dummy, you will have the bug; all the time. Now for example if a dudu(boomkin) invites you to a group, you will see that you get a haste buff. If you try your rotation now, you will notice there is no difference and the bug is still there. Now ask the dudu to reactivate his passive (change to normal form and back to boomkin form) Now do your rotation and you have no more bug.. Now idk how it come but it works on everything which gives you a haste buff, an elixer of that 225 haste for 1 hour works, any other haste buff which you can rebuff (like shadowpriests and boomkins by changing state) the bug is no longer there. If you do not have such a kind of haste raid/party buff time warp also works to unbug it. Know that you will get the bug back whenever time warp is over. I don't really know how this is fixing the problem but i noticed it after i did a dungeon where there was a shadowpriest in it.

**Comment by Iambugged** [comment]
I'm having this same problem while I have the 9% attack speed buff from Judgement of the Pure talent active but without of that it doesn't seem to happen.

**Comment by Frainzil** [comment]
I tested that talent also, but it doesn't really make any difference for me. The only thing that works is using or a haste pot or having a raid haste buff getting re-activated.

**Comment by Frainzil** [comment]
I'm no longer having problems with this bug, maybe it got fixed somehow, maybe it's just temporarely.

**Comment by Malvoli0** [comment]
I still have this problem. I am not sure why it happens but it seems that the cooldown of Crusader Strike / Divine Storm isn't reduced by some cd reducing talent like it should be, this creating a difference in how your client and the server view the cooldown.

**Comment by Jenovas** [comment]
Well I also did some tests on my home server and I even added a aura to myself that gives 100 000% haste making CS have no cooldown (or cooldown = gcd) and even when spamming it seemed to work ok.
Maybe the problem is latency? Maybe sometimes the server and client seem to fall out of synchro? I'll be testing this again but this time on our server, hope to find the bug.

**Comment by Clarkiez96** [comment]
I thought this was fixed ages ago? Not had a problem for a very long time

**Comment by Jenovas** [comment]
Malvoli0 claims that the problem still exists, but I cant seem to get it to bug.

**Comment by Clarkiez96** [comment]
Falls claims then, at 20ms with the server I have no problems and then using a weaker connection causing me to have 140ms I had no problems

**Comment by Clarkiez96** [comment]
False*

**Comment by sunshineinabox** [comment]
I am experiencing a similar issue with Scorch the fire mage talent. Not sure if it is related. Should I make a separate issue for that?

**Comment by Malvoli0** [comment]
False claims ? If you aren't experiencing this issue, then I don't see why you would derail my efforts to see it fixed. 

I play at 39-40ms latency, I can't see it being related to latency since the ping form the server is the one that returns "still on cooldown" back, your client can't do that on it's own. Besides, this i sth eonly ability I'm experiencing the issue with. I guess I could fraps it so you can for yourself I'm not lying.

**Comment by Clarkiez96** [comment]
Fraps it then, there has been no reports for about 2 months

**Comment by Frainzil** [comment]
clarkie, i frapsed it 2 months ago, had serious problems than, Though this bug got fixed for me everytime i had a raid-haste buff. Moonkin aura, shadowpriest, totem of shaman... Eventually the bug disappeared, Now i JUST did a test i was actually thinking about of what is causing. I removed several items that give me haste (i had around 5,5% haste of items), until my haste% is 1,34%. I did a dummy test again and tada, bug appears. SO i think everyone with lower gear, who doesn't have a certain haste ''cap'' have the bug, everyone who has more haste = no bug. How this can happen? No idea
